velox/exec/HashTable.cpp

```diff
@@ -620,7 +620,8 @@ void HashTable<ignoreNullKeys>::checkSize(int32_t numNew) {
     // numDistinct_ is non-0 when switching from HashMode::kArray to regular
     // hashing.
     auto newSize = std::max(
-        (uint64_t)2048, bits::nextPowerOfTwo(numNew * 2 + numDistinct_));
+        (uint64_t)2048,
+        bits::nextPowerOfTwo(numNew * 2 + numDistinct_ + numDistinct_ / 8));
```

Comment:
What's the rational for numDistinct_ / 8 ?

Would you update the PR description to explain the problem and the chosen solution?

Please, add a test.

Reply:
+1